src/jdk.hotspot.agent/linux/native/libsaproc/LinuxDebuggerLocal.cpp

```diff
@@ -345,12 +345,14 @@ JNIEXPORT jlong JNICALL Java_sun_jvm_hotspot_debugger_linux_LinuxDebuggerLocal_l
   jlong addr;
   jboolean isCopy;
   struct ps_prochandle* ph = get_proc_handle(env, this_obj);
-  AutoJavaString objectName_cstr(env, objectName);
-  CHECK_EXCEPTION_(0);
+  // Note, objectName is ignored, and may in fact be NULL.
+  // lookup_symbol will always search all objects/libs
+  //AutoJavaString objectName_cstr(env, objectName);
```

Comment:
I think it would be better to update AutoJavaString class to handle null strings:
`m_buf(str == NULL ? NULL : env->GetStringUTFChars(str, NULL)) `

Reply:
The point is the argument is ignored, so it doesn't really matter if AutoJavaString can handle NULL. I  assume with your suggested fix that also want me to pass objectName_cstr to lookup_symbol() rather than an explicit NULL. I wanted the NULL to be explicit to help further convey that objectName is ignored.

Reply:
lookup_symbol implementation contains FIXME comment about object_name.
I suppose you want to make this argument ignorance permanent. Then I think this FIXME should be removed (or updated) and it would be nice to add comment about this to lookup_symbol declaration (or maybe it would be better to delete this argument)

Reply:
If the FIXME in `lookup_symbol` was ever addressed, then we would need for NULL to mean to search all libraries. I suppose in that case it would make sense for AutoJavaString to do as you suggested. It looks like Windows also has an AutoJavaString. I'll change it also, but it is never passed NULL. The Bsd version is, but it handles the GetStringUTFChars() inline without AutoJavaString, and it does support NULL already:

```
  objectName_cstr = NULL;
  if (objectName != NULL) {
    objectName_cstr = (*env)->GetStringUTFChars(env, objectName, &isCopy);
    CHECK_EXCEPTION_(0);
  }
...
  addr = (jlong) lookup_symbol(ph, objectName_cstr, symbolName_cstr);

```

Reply:
I've pushed the AutoJavaString change.